pkg/app/postgres-pgbasebackup.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2019 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/helm"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/log"
+	"k8s.io/client-go/kubernetes"
+)
+
+type PostgresSQLDB struct {
+	name      string
+	cli       kubernetes.Interface
+	chart     helm.ChartInfo
+	namespace string
+}
+
+const (
+	postgresUser = "postgres"
+	waitToCount  = 2 * time.Minute
+)
+
+func NewPostgresSQLDB(name string) App {
+	return &PostgresSQLDB{
+		name: name,
+		chart: helm.ChartInfo{
+			Release:  appendRandString(name),
+			Chart:    "postgresql",
+			RepoName: helm.StableRepoName,
+			RepoURL:  helm.StableRepoURL,
+			Version:  "8.6.4",
+			Values: map[string]string{
+				"image.repository": "kanisterio/postgresql",
+				"image.tag":        "0.28.0",
+				"pgHbaConfiguration": `host     all             all             0.0.0.0/0               trust
```

Comment:
minor: this is hard to read, please define a constant and remove some of the extra spaces